include/fmt/core.h

```diff
@@ -468,6 +468,18 @@ class value {
     set<POINTER>(pointer, p);
   }
 
+  // Match function pointers explicitly. If we do not then the
+  // forwarding T * constructor below will end up trying to call
+  // itself when a function pointer is passed, which leads to a
+  // confusing error message.
+  template <typename T, typename ...Args>
+  value(T (*p)(Args...)) {
+    // This static_assert must use something that is dependent on T,
+    // otherwise it will always fire, so let's use the same check as
+    // above - it will never match.
+    static_assert(std::is_same<T, void>::value, "formatting of function pointers is disallowed");
```

Comment:
Could you use `FMT_STATIC_ASSERT()` here instead?

Reply:
Thanks for the reply.

I copied the constructor above which uses static_assert. There is no mention of FMT_STATIC_ASSERT on the std branch. I can only assume that the std branch is targeting modern C++ versions that support static_assert.

I think that using std::is_pointer along with std::is_function and then std::is_member would probably produce clearer code. I'll wait for @vitaut to provide his opinion before having a go at that though.

Reply:
Oh, right. You're on the std branch. In that case you're good.